File: app/src/main/java/bogomolov/aa/fitrack/TrackerApplication.java
package bogomolov.aa.fitrack;

import android.app.Application;

import javax.inject.Inject;

import bogomolov.aa.fitrack.dagger.AppComponent;
import dagger.android.AndroidInjector;
import dagger.android.DispatchingAndroidInjector;
import dagger.android.HasAndroidInjector;

import bogomolov.aa.fitrack.dagger.DaggerAppComponent;


public class TrackerApplication extends Application implements HasAndroidInjector {

    @Inject
    DispatchingAndroidInjector<Object> androidInjector;


    @Override
    public void onCreate() {
        super.onCreate();

        DaggerAppComponent.builder().application(this).build().inject(this);
    }

    @Override
    public AndroidInjector<Object> androidInjector() {
        return androidInjector;
    }
}

File: app/src/main/java/bogomolov/aa/fitrack/TrackerApplication.kt
package bogomolov.aa.fitrack

import android.app.Application
import bogomolov.aa.fitrack.di.buildModules
import org.koin.android.ext.koin.androidContext
import org.koin.android.ext.koin.androidLogger
import org.koin.core.Koin
import org.koin.core.context.startKoin

class TrackerApplication : Application() {

    lateinit var koin: Koin

    override fun onCreate() {
        super.onCreate()
        koin = startKoin {
            androidLogger()
            androidContext(this@TrackerApplication)
            modules(buildModules())
        }.koin
    }
}
